pandas/tests/io/test_sql.py

```diff
@@ -4331,3 +4330,35 @@ def test_xsqlite_if_exists(sqlite_buildin):
         (5, "E"),
     ]
     drop_table(table_name, sqlite_buildin)
+
+
+def test_execution_of_multi(mysql_pymysql_engine):
```

Comment:
This test should test a public API like

```python
df = DataFrame(...)
df.to_sql(..., method="multi")
result = pd.read_sql(...)
tm.assert_frame_equal(df, result)

```

Reply:
When you say this, do you mean that I should test that the data inserted matches the data read. The test that I performed shows that the statement executed is multi-value. Without a multi-value insert, all the data is still executed, it is just executed as multiple SQL statements. Here I show that 1 SQL statement contains multiple rows.

Reply:
OK I see the difficulty. I would still say we don't want to add _this_ complex of a unit test since it can become hard to debug/maintain in the future.

Let's just remove this unit test for now. I think it will take some reorganizing of the sql code to make testing this easier

Reply:
Is the complexity in the monkeypatching or in the regex of the SQL statement? Because perhaps I can find some other way to analyze the statement that is not using regex?
I added a similar unit test on an internal repo which writes to a PrestoDB. There, however, I justed added a timeout because one was exponentially longer than the other.

Reply:
I simplified the monkeypatch -- I think this is a pretty durable test now.

Reply:
Any reason to not use the `sqlalchemy_connectable` fixture for this?

Also I agree with @mroeschke - this test gets into the internals of both pandas implementation and event listening in SQLAlchemy. I don't think there is value in forcing developers / maintainers to have that detailed of knowledge of both.

Reply:
@WillAyd The sqlalchemy_connectable would work well.
@mroeschke So is the move to not do a unit test here?

Reply:
Yes I would still recommend not having a unit test for now.

Reply:
How do it look now?